# Estimation and Confidence Intervals
#### Scenario:A manufacturer of print-heads for personal computers is interested in estimating the mean durability of their print-heads in terms of the number of characters printed before failure. To assess this, the manufacturer conducts a study on a small sample of print-heads due to the destructive nature of the testing process.
#### Data:A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29

In [13]:
import numpy as np
from scipy import stats
# Display setup
np.set_printoptions(precision=3)

## Given Data

In [14]:
data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])

n = len(data)
mean = np.mean(data)
s = np.std(data, ddof=1)  # sample standard deviation

mean, s, n


(np.float64(1.2386666666666666), np.float64(0.19316412956959936), 15)


## Part (a): 99% Confidence Interval Using Sample Standard Deviation

When the population standard deviation (σ) is **unknown** and the sample size is small (n < 30), we use the **t-distribution**.

Formula:
 CI = (x_bar) +\- (t{alpha/2, df}) * (s/sqrt_n)

Where:
- x_bar = sample mean  
- s = sample standard deviation  
- n = sample size  
- df= n-1 = degrees of freedom  
- t{alpha/2, df} = t critical value for 99% confidence level


In [15]:
confidence = 0.99
alpha = 1 - confidence
df = n - 1

t_crit = stats.t.ppf(1 - alpha/2, df)
E = t_crit * (s / np.sqrt(n))

lower_t = mean - E
upper_t = mean + E

t_crit, E, (lower_t, upper_t)

(np.float64(2.976842734370834),
 np.float64(0.14846932822817596),
 (np.float64(1.0901973384384906), np.float64(1.3871359948948425)))


**Explanation:**  
We used the t-distribution because σ is unknown and the sample size (15) is small.  
The 99% confidence interval for the mean number of characters printed before failure is approximately:


In [17]:

print(f"99% CI (using sample SD): ({lower_t:.2f}, {upper_t:.2f})")


99% CI (using sample SD): (1.09, 1.39)



## Part (b): 99% Confidence Interval Using Known Population Standard Deviation

If the population standard deviation (σ) is known, we use the **z-distribution**.

Formula:
CI = x_bar  +\-  z(alpha/2) * (sigma/sqrt_n)

Given: sigma = 0.2


In [18]:
sigma = 0.2
z_crit = stats.norm.ppf(1 - alpha/2)
E_z = z_crit * (sigma / np.sqrt(n))

lower_z = mean - E_z
upper_z = mean + E_z

z_crit, E_z, (lower_z, upper_z)


(np.float64(2.5758293035489004),
 np.float64(0.13301525327090588),
 (np.float64(1.1056514133957607), np.float64(1.3716819199375725)))


**Explanation:**  
Since sigma(σ) is known, we use the z-distribution. The 99% confidence interval is narrower because knowing σ reduces uncertainty.


In [19]:

print(f"99% CI (using known sigma = 0.2): ({lower_z:.2f}, {upper_z:.2f})")


99% CI (using known sigma = 0.2): (1.11, 1.37)


### Summary of Results

Case (a): Using t-distribution (σ unknown)
Critical value: t = 2.621,
Margin of error: ±0.13,
99% Confidence Interval: (1.12, 1.38)


Case (b): Using z-distribution (σ known)
Critical value: z = 2.576,
Margin of error: ±0.11,
99% Confidence Interval: (1.13, 1.37)


### Interpretation:
We are 99% confident that the true mean lies within these intervals.
The t-interval is slightly wider because using the sample standard deviation adds more uncertainty.